bulk/utils.py

```diff
@@ -20,6 +20,9 @@ def get_color_mapping(df: pd.DataFrame) -> Tuple[Optional[bokeh.transform.transf
         all_values = list(df["color"].dropna().unique())
         if len(all_values) == 2:
             all_values.extend([""])
+        elif len(all_values) > len(Category10) + 2:
```

Comment:
I felt a bit weird hard-coding the limit, but not sure if this is better, since if you were to change the palette you'd still need to manually change this as well..

Reply:
I think this is fine for now though. The project is in the "throw it on the wall and see what sticks" phase. Later I'll be adding support for images as well and after that's been out for a while it'll be time to think about proper abstractions.